### ISSUE TITLE
[Feature] Notification

### ISSUE THREAD

**Issue by plagioriginal** [lardbit/nefarious#26]
Thank you for this cool project! <3

Maybe somekind of notification saying that a download started, maybe by email or pushbullet ou gotify whould be nice?

Thanks!

**Comment by lardbit** [comment]
I think that's a great idea.  I'd definitely like to see this feature.  I'll research some options when I get some time.

**Comment by DevAndrei** [comment]
Support for webhooks would be nice

**Comment by lardbit** [comment]
I like the webhooks idea. I was also thinking of using push notifications. Thoughts?

**Comment by lardbit** [comment]
#45 addresses this with webhook implementation.  Expects webhook to accept `json` data.  You specify the webhook url and data key.

**Comment by plagioriginal** [comment]
Hey @lardbit ,
sorry to bother with this again.

Can you add a field in which I can specify a "priority" in the json object that you send to the webhook? My push notification works with a "priority" field that we can put from 0 to 10, and the way it works currently it only goes for a 0, I would like to put it at least a 5.
I know that this a not so "generic" request, sorry for that. I already tried to hard code it into the webhook.py in my local, but has I am using docker-compose.yml it's not working properly, it never adds the "priority" to the object

Thanks

**Comment by lardbit** [comment]
Ok, so you just want to add a new key `priority` with a value of `5` to the webhook post?  You're right in that it's not a generic request, but, what I could do instead would be to let users add custom json to the request.  So I could add an input box where you could enter some json like the following:


```json
{
  "priority": 5,
}
```

and it would simply pass that along as well to the webhook.  

I should probably let users define the entire payload/body of the json and use parameter substitution to it which should work great with everyone.

For instance, you could enter something like the following to the webhook input box:

```json
{
  "message": "Media [[media_title]] was successfully downloaded",
  "priority": 5,
}
```
and `media_title` would be replaced by the actual thing that was downloaded.

Thoughts?

**Comment by plagioriginal** [comment]
Hey,

The first solution would work perfectly in my case, but maybe allowing the entire body of the json to be changes would be more flexible.

I would settle for the thing that gives you less development work :)

Thanks!

**Comment by lardbit** [comment]
Sounds good.  I'll reopen this issue and update when it's complete.

**Comment by lardbit** [comment]
nefarious integrated notifications via apprise which supports all major platforms. Hopefully that resolves this.

https://github.com/caronc/apprise
https://github.com/lardbit/nefarious/pull/151

**Comment by panmau** [comment]
I am trying to setup the notifications with Gotify but I receive "Notification returned an error" and in the logs of the nefarious docker  container I see: `Failed to send Gotify notification: Bad Request - Unsupported Parameters., error=400.`
I use an url in the following format: `gotifys://{hostname}/{token}` as described in the Apprise [documentation](https://github.com/caronc/apprise/wiki/Notify_gotify).

What else can I try to make this work? Can you maybe try whether it works for you?

**Comment by lardbit** [comment]
Is your `hostname` equal to **localhost**?  nefarious is run via docker containers which means it's operating in a private network mode where it wouldn't be able to see a local instance of *gotify* unless you're running it in the same `docker-compose.yml` stack as nefarious.  Is your `hostname` set to **localhost**?  How/where is gotify running?

**Comment by lardbit** [comment]
I recommend editing the `docker-compose.yml` file within nefarious and adding gotify, and then you would be able to define the **Notification URL** as something like:

    gotify://gotify:8080/your-token

Where `gotify` is the name of the docker service you define, and it's mapped to port `8080`

**Comment by panmau** [comment]
> Is your `hostname` equal to **localhost**?  nefarious is run via docker containers which means it's operating in a private network mode where it wouldn't be able to see a local instance of *gotify* unless you're running it in the same `docker-compose.yml` stack as nefarious.  Is your `hostname` set to **localhost**?  How/where is gotify running?

No, my instance is run on a server outside my local network and reachable by subdomain.domain.ending which I set as my hostname.

**Comment by lardbit** [comment]
got it. So what happens when you try and test it out via:

    apprise -vv -t 'my title' -b 'my notification body' gotifys://{hostname}/{token}

That should at least tell you if the communication from apprise to your gotify server is working.

**Comment by panmau** [comment]
> got it. So what happens when you try and test it out via:
> 
> ```
> apprise -vv -t 'my title' -b 'my notification body' gotifys://{hostname}/{token}
> ```
> 
> That should at least tell you if the communication from apprise to your gotify server is working.

Good idea. I tried it out and verified that it works successfully:
![image](https://user-images.githubusercontent.com/16801528/141684936-40117444-0967-4b96-b6bd-9421b74d49c7.png)

**Comment by lardbit** [comment]
Ok, I'll setup a gotify server and test this as well.  I'm using apprise + Slack so I know that's working, but maybe there's other things to consider.  I'l get back to you.

**Comment by lardbit** [comment]
You're right.  I was only sending the message `title` without the `body`, but I guess each service provider is different and I should have just been sending `body`.  I updated it and it's working with my gotify server correctly.

I committed https://github.com/lardbit/nefarious/commit/0114761ff05619b747766c8f2480bc696df5ba02 so give it an hour or so to build out and update your installation.

Thanks for the feedback.
